root/edit/details/AddRelationshipType.js

```diff
@@ -0,0 +1,147 @@
+/*
+ * @flow
+ * Copyright (C) 2020 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import * as React from 'react';
+
+import Cardinality from '../../static/scripts/common/components/Cardinality';
+import OrderableDirection
+  from '../../static/scripts/common/components/OrderableDirection';
+import {ENTITY_NAMES} from '../../static/scripts/common/constants';
+import expand2react from '../../static/scripts/common/i18n/expand2react';
+
+type AddRelationshipTypeEditT = {
+  ...EditT,
+  +display_data: {
+    +attributes: $ReadOnlyArray<{
+      ...LinkTypeAttrTypeT,
+      +typeName: string,
+    }>,
+    +child_order: number,
+    +description: string | null,
+    +documentation: string | null,
+    +entity0_cardinality?: number,
+    +entity0_type: CoreEntityTypeT,
+    +entity1_cardinality?: number,
+    +entity1_type: CoreEntityTypeT,
+    +link_phrase: string,
+    +long_link_phrase: string,
+    +name: string,
+    +orderable_direction?: number,
+    +reverse_link_phrase: string,
+  },
+};
+
+type Props = {
+  +edit: AddRelationshipTypeEditT,
+};
+
+const AddRelationshipType = ({edit}: Props): React.Element<'table'> => {
+  const display = edit.display_data;
+  const entity0Type = ENTITY_NAMES[display.entity0_type]();
+  const entity1Type = ENTITY_NAMES[display.entity1_type]();
+
+  return (
+    <table className="details add-relationship-type">
+      <tr>
+        <th>{addColonText(l('Name'))}</th>
+        <td>{display.name}</td>
+      </tr>
+      <tr>
+        <th>{addColonText(l('Description'))}</th>
+        <td>
+          {display.description
+            ? expand2react(display.description)
+            : lp('(none)', 'description')}
+        </td>
+      </tr>
+      <tr>
+        <th>{l('Link phrase:')}</th>
+        <td>
+          <em>{entity0Type}</em>
+          {' '}
+          {display.link_phrase}
+          {' '}
+          <em>{entity1Type}</em>
+        </td>
+      </tr>
+
+      <tr>
+        <th>{l('Reverse link phrase:')}</th>
+        <td>
+          <em>{entity1Type}</em>
+          {' '}
+          {display.reverse_link_phrase}
+          {' '}
+          <em>{entity0Type}</em>
+        </td>
+      </tr>
+
+      <tr>
+        <th>{l('Long link phrase:')}</th>
+        <td>
+          <em>{entity0Type}</em>
+          {' '}
+          {display.long_link_phrase}
+          {' '}
+          <em>{entity1Type}</em>
+        </td>
+      </tr>
```

Comment:
It is barely readable, I would suggest to just keep `{entity0}` and `{entity1}` in link phrases and to add two fields: entity0 type and entity1 type. That would also help with interpreting cardinality fields correctly.

Reply:
Hmm. I'm just restoring what was the intended display of this, what do you find barely readable about it? :) 
I expect for most people seeing such an edit, it's much more clear to see "artist recorded release" than "entity0 recorded entity1"...

Reply:
It sometimes work in English but not always. Two counter-examples:

* http://localhost:5000/edit/53062894
  > Link phrase: _Label_ get the music _URL_

* http://localhost:5000/edit/52510291
  > Long link phrase: _Artist_ {entity1} was premiered by {entity0} _Work_

Plus, I suspect that inserting entity type names like that might not work as “well” in some other languages.

Reply:
The fun thing here is that I'm not dropping entity0 and entity1 from the link phrase, it just works like that unless it is specifically changed :) But I guess I can just leave it as it was (showing only the phrase) and display the types separately, sure!